switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,678 @@
+    if version_minor > VERSION_MINOR:
+        logger.debug("Provided Minor version is upper than current one, new"
+                     "optional functionality may not be handled in this"
+                     "script")
+
+    for sriov_pf in config.get("pfs"):
+        if sriov_pf.get("name"):
+            sriov_pfs_list.append(sriov_pf.get("name"))
+        else:
+            err = "sriov pf name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        try:
+            if int(sriov_pf.get("numOfVfs")):
+                continue
+        except ValueError as value_err:
+            logger.error("numOfVfs for %s must be number" %
+                         sriov_pf.get("name"))
+            raise value_err
+        except TypeError as type_err:
+            logger.error("numOfVfs for %s must be provided" %
+                         sriov_pf.get("name"))
+    for bond in config.get("linuxBonds"):
+        if bond.get("name"):
+            bonds_list.append(bond.get("name"))
+        else:
+            err = "bond name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if not bond.get("bondingOptions"):
+            err = "bondingOptions must be provided for bond %s" %\
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+        if (bond.get("slaves") and
+                (set(bond.get("slaves")).issubset(set(sriov_pfs_list)))):
+            continue
+        else:
+            err = "slaves of bond %s must be list of sriov_pfs" % \
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+    for bridge in config.get("ovsBridges"):
+        if not bridge.get("name"):
+            err = "bridge name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if (bridge.get("ports") and
+                (set(bridge.get("ports")).issubset(set(sriov_pfs_list +
+                                                   bonds_list)))):
+            continue
+        else:
+            err = "ports of bridge %s must be list of sriov_pfs or bonds" % \
+                  bridge.get("name")
+            logger.error(err)
+            raise ValueError(err)
+
+
+def parse_opts(argv):
+
+    parser = argparse.ArgumentParser(
+        description='SR-IOV, OVS and network configuration service'
+        'for switchdev and VF-LAG configuration')
+
+    parser.add_argument(
+        '--sriov_bind',
+        action="store_true",
+        help="Do sriov vfs bind for sriov_bind service in reboot",
+        required=False)
+
+    parser.add_argument(
+        '--sriov_config',
+        action="store_true",
+        help="Do sriov configuration for sriov_config service in reboot",
+        required=False)
+
+    parser.add_argument(
+        '-v', '--version',
+        action='version',
+        version="%d.%d" % (VERSION_MAJOR, VERSION_MINOR))
```

Comment:
I would really prefer a Major.Minor.SubMinor version

so that:
Major: marks a breaking change
Minor: is a release with new features
SubMinor is generally bugfixes to current release with no new features added